ddlambda.go

```diff
@@ -42,54 +43,65 @@ type (
+	// The following constants specify the environment variables that will...
+	// ...set the API key
 	DatadogAPIKeyEnvVar = "DD_API_KEY"
-	// DatadogKMSAPIKeyEnvVar is the environment variable that will be sent to KMS for decryption, then used as an API key.
+	// ...be sent to KMS for decryption, then used as an API key.
 	DatadogKMSAPIKeyEnvVar = "DD_KMS_API_KEY"
-	// DatadogSiteEnvVar is the environment variable that will be used as the API host.
+	// ...be used as the API host.
 	DatadogSiteEnvVar = "DD_SITE"
-	// DatadogLogLevelEnvVar is the environment variable that will be used to check the log level.
+	// ...be used to check the log level.
 	// if it equals "debug" everything will be logged.
 	DatadogLogLevelEnvVar = "DD_LOG_LEVEL"
-	// DatadogShouldUseLogForwarderEnvVar is the environment variable that is used to enable log forwarding of metrics.
+	// ...be used to enable log forwarding of metrics.
 	DatadogShouldUseLogForwarderEnvVar = "DD_FLUSH_TO_LOG"
+	// ...be used to check if DD tracing is enabled.
+	DatadogTraceEnabledEnvVar = "DD_TRACE_ENABLED"
+	// ...be used to check if X-Ray traces should be merged
+	DatadogMergeXrayTracesEnvVar = "DD_MERGE_XRAY_TRACES"
+
 	// DefaultSite to send API messages to.
 	DefaultSite = "datadoghq.com"
 	// DefaultEnhancedMetrics enables enhanced metrics by default.
 	DefaultEnhancedMetrics = true
 )
 
-// WrapHandler is used to instrument your lambda functions, reading in context from API Gateway.
-// It returns a modified handler that can be passed directly to the lambda.Start function.
+// WrapHandler is used to instrument your lambda functions.
+// It returns a modified handler that can be passed directly to the lambda. Start function.
 func WrapHandler(handler interface{}, cfg *Config) interface{} {
 
 	logLevel := os.Getenv(DatadogLogLevelEnvVar)
 	if strings.EqualFold(logLevel, "debug") || (cfg != nil && cfg.DebugLogging) {
 		logger.SetLogLevel(logger.LevelDebug)
 	}
 
-	// Set up state that is shared between handler invocations
-	tl := trace.Listener{}
+	// Wrap the handler with listeners that add instrumentation for traces and metrics.
+	tl := trace.MakeListener(cfg.toTraceConfig())
 	ml := metrics.MakeListener(cfg.toMetricsConfig())
 	return wrapper.WrapHandlerWithListeners(handler, &tl, &ml)
 }
 
-// GetTraceHeaders reads a map containing the DataDog trace headers from a context object.
+// GetTraceHeaders reads a map containing the Datadog trace headers from a context object.
+// Deprecated: Use dd-trace-go to extract the current span from the context instead
 func GetTraceHeaders(ctx context.Context) map[string]string {
 	result := trace.GetTraceHeaders(ctx, true)
 	return result
 }
 
-// AddTraceHeaders adds DataDog trace headers to a HTTP Request
+// AddTraceHeaders adds Datadog trace headers to a HTTP Request
+// Deprecated: Use dd-trace-go to extract the current span from the context instead
```

Comment:
@DarcyRaynerDD should we totally remove these methods (`GetTraceHeaders` and `AddTraceHeaders`)? I think they will basically not work any more because they return the headers representing the trace context within which the Lambda execution takes place, but not the Lambda execution span itself. So if these headers are used for propagation, the trace might look weird/wrong. Also, we might be able to do some more simplification if we no longer need to worry about propagating headers in this library, and just let `dd-trace-go` handle everything related to that.

On the other hand, removing these would obviously be a breaking change.

Reply:
I think marking it as deprecated is fine, as long as it still works when we are just using xray without dd-trace